In [1]:
import pandas as pd
from sklearn.datasets import load_iris

data = load_iris(as_frame=True)
df = data.frame


print("Shape:", df.shape)
display(df.head())
print("Columns:", df.columns)
df.info()

Shape: (150, 5)


sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                5.1               3.5                1.4               0.2   
1                4.9               3.0                1.4               0.2   
2                4.7               3.2                1.3               0.2   
3                4.6               3.1                1.5               0.2   
4                5.0               3.6                1.4               0.2   

   target  
0       0  
1       0  
2       0  
3       0  
4       0

Columns: Index(['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)',
       'petal width (cm)', 'target'],
      dtype='object')
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
 4   target             150 non-null    int64  
dtypes: float64(4), int64(1)
memory usage: 6.0 KB


Problem Type: Classification

Goal:
Predict the type of iris flower using sepal and petal measurements.

Target Variable:
target

Description:
The dataset contains measurements of flowers. The model will learn to classify the flower into one of three species.

In [2]:
print("Target values counts:")
print(df["target"].value_counts())

Target values counts:
target
0    50
1    50
2    50
Name: count, dtype: int64


Dataset Summary:
Dataset Name: Iris Dataset (from scikit-learn)
Type: Tabular (rows and columns)
Size: 150 rows × 5 columns
Features: sepal length, sepal width, petal length, petal width
Target: target (3 classes representing iris species)